src/commonMain/kotlin/fr/acinq/lightning/bin/csv/WalletPaymentCsvWriter.kt

```diff
@@ -0,0 +1,126 @@
+package fr.acinq.lightning.bin.csv
+
+import fr.acinq.bitcoin.ByteVector32
+import fr.acinq.bitcoin.Satoshi
+import fr.acinq.bitcoin.TxId
+import fr.acinq.lightning.MilliSatoshi
+import fr.acinq.lightning.db.*
+import fr.acinq.lightning.utils.msat
+import fr.acinq.lightning.utils.sat
+import fr.acinq.lightning.utils.sum
+import fr.acinq.lightning.utils.toMilliSatoshi
+import kotlinx.datetime.Instant
+import okio.Path
+
+class WalletPaymentCsvWriter(path: Path) : CsvWriter(path) {
+
+    private val FIELD_DATE = "date"
+    private val FIELD_TYPE = "type"
+    private val FIELD_AMOUNT_MSAT = "amount_msat"
+    private val FIELD_FEE_CREDIT_MSAT = "fee_credit_msat"
+    private val FIELD_MINING_FEE_SAT = "mining_fee_sat"
+    private val FIELD_SERVICE_FEE_MSAT = "service_fee_msat"
+    private val FIELD_PAYMENT_HASH = "payment_hash"
+    private val FIELD_TX_ID = "tx_id"
+
+    init {
+        addRow(FIELD_DATE, FIELD_TYPE, FIELD_AMOUNT_MSAT, FIELD_FEE_CREDIT_MSAT, FIELD_MINING_FEE_SAT, FIELD_SERVICE_FEE_MSAT, FIELD_PAYMENT_HASH, FIELD_TX_ID)
+    }
+
+    data class Details(
+        val type: String,
+        val amount: MilliSatoshi,
+        val feeCredit: MilliSatoshi,
+        val miningFee: Satoshi,
+        val serviceFee: MilliSatoshi,
+        val paymentHash: ByteVector32?,
+        val txId: TxId?
+    )
+
+    private fun addRow(
+        timestamp: Long,
+        details: Details
+    ) {
+        val dateStr = Instant.fromEpochMilliseconds(timestamp).toString() // ISO-8601 format
+        addRow(
+            dateStr,
+            details.type,
+            details.amount.msat.toString(),
+            details.feeCredit.msat.toString(),
+            details.miningFee.sat.toString(),
+            details.serviceFee.msat.toString(),
+            details.paymentHash?.toHex() ?: "",
+            details.txId?.toString() ?: ""
+        )
+    }
+
+    fun add(payment: WalletPayment) {
```

Comment:
@t-bast This function is what should mainly be reviewed. I've done a good amount of testing in various scenarii but you never know.